src/ui/public/vis/components/tooltip/tooltip.js

```diff
@@ -113,16 +140,48 @@ Tooltip.prototype.$getChart = function () {
 Tooltip.prototype.render = function () {
   const self = this;
 
+  const mouseEnter = _.debounce(function (d3Event, d, i) {
+    if (self.enterCount === 0) {
+      return;
+    }
+    const currentD3Event = d3.event;
+    d3.event = d3Event;
+    if (!self.showCondition(d, i)) {
+      self.contentContainer.empty();
+    } else {
+      const events = self.events ? self.events.eventResponse(d, i) : d;
+      self.contentContainer.html(self.formatter(events));
```

Comment:
I'd consider changing this `self.formatter` method instead of returning a html-string, to having it return a DOM-node, or append to a DOM-node.

This will avoid having to create dynamically generate IDs and then querying the DOM for the container-node because you can just keep a reference to the node directly. (see e.g. the work-around of creating the `executionId` just so we can find the node in embeddedTooltipFormatter).

Reply:
As discussed, this is a good idea but should be done outside of the scope of this PR